components/Banner/tests/unit/BannerViewTests.m

```diff
@@ -0,0 +1,100 @@
+
+@interface MDCBannerViewTestsDynamicTypeContentSizeCategoryOverrideWindow : UIWindow
+
+/** Used to override the value of @c preferredContentSizeCategory. */
+@property(nonatomic, copy) UIContentSizeCategory contentSizeCategoryOverride;
+
+@end
+
+@implementation MDCBannerViewTestsDynamicTypeContentSizeCategoryOverrideWindow
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = UIContentSizeCategoryLarge;
+  }
+  return self;
+}
+
+- (instancetype)initWithContentSizeCategoryOverride:
+    (UIContentSizeCategory)contentSizeCategoryOverride {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = contentSizeCategoryOverride;
+  }
+  return self;
+}
+
+- (UITraitCollection *)traitCollection {
+  if (@available(iOS 10.0, *)) {
+    UITraitCollection *traitCollection = [UITraitCollection
+        traitCollectionWithPreferredContentSizeCategory:self.contentSizeCategoryOverride];
+    return traitCollection;
+  }
+  return [super traitCollection];
+}
+
+@end
+
+@interface BannerViewTests : XCTestCase
+
+@end
+
+@implementation BannerViewTests
+
+- (void)testBannerViewDynamicTypeBehavior {
+  if (@available(iOS 10.0, *)) {
+    // Given
+    MDCBannerView *bannerView = [[MDCBannerView alloc] init];
+    bannerView.trailingButton.hidden = YES;
+    bannerView.mdc_adjustsFontForContentSizeCategory = YES;
+    UIFont *font = [UIFont systemFontOfSize:10.0 weight:UIFontWeightRegular];
+    MDCFontScaler *fontScaler = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody2];
+    UIFont *scalableFont = [fontScaler scaledFontWithFont:font];
+    scalableFont = [scalableFont mdc_scaledFontAtDefaultSize];
+    bannerView.textLabel.font = scalableFont;
+    bannerView.textLabel.text = @"Banner Text";
+    CGFloat originalTextFontSize = bannerView.textLabel.font.pointSize;
+    MDCButton *leadingButton = bannerView.leadingButton;
+    [leadingButton setTitleFont:scalableFont forState:UIControlStateNormal];
+    CGFloat originalButtonFontSize =
+        [bannerView.leadingButton titleFontForState:UIControlStateNormal].pointSize;
+
+    // When
+    MDCBannerViewTestsDynamicTypeContentSizeCategoryOverrideWindow *extraExtraLargeContainer =
+        [[MDCBannerViewTestsDynamicTypeContentSizeCategoryOverrideWindow alloc]
+            initWithContentSizeCategoryOverride:UIContentSizeCategoryExtraExtraLarge];
+    [extraExtraLargeContainer addSubview:bannerView];
+    [NSNotificationCenter.defaultCenter
+        postNotificationName:UIContentSizeCategoryDidChangeNotification
+                      object:nil];
+
+    // Then
+    CGFloat actualTextFontSize = bannerView.textLabel.font.pointSize;
+    XCTAssertGreaterThan(actualTextFontSize, originalTextFontSize);
+    CGFloat actualButtonFontSize =
+        [bannerView.leadingButton titleFontForState:UIControlStateNormal].pointSize;
+    XCTAssertGreaterThan(actualButtonFontSize, originalButtonFontSize);
+  }
+}
+
+@end
```

Comment:
Are we able to add snapshot tests for this behavior? They may be preferable to logic tests in this case.

Reply:
Added in f428e8e7b6110f0c28bb638c8a9317983a74048d